src/internal/packager/template/template_test.go

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package template provides functions for templating yaml files.
+package template
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/zarf-dev/zarf/src/pkg/variables"
+)
+
+func TestGetSanitizedTemplateMap(t *testing.T) {
+	t.Parallel()
+	input := map[string]*variables.TextTemplate{
+		"###ZARF_GIT_AUTH_PULL###": {Sensitive: true, Value: "secret1"},
+		"###ZARF_GIT_AUTH_PUSH###": {Sensitive: true, Value: "secret2"},
+		"###ZARF_REGISTRY###":      {Sensitive: false, Value: "127.0.0.1:31999"},
+		"###ZARF_GIT_PUSH###":      {Sensitive: false, Value: "zarf-git-user"},
+		"###ZARF_GIT_PULL###":      {Sensitive: false, Value: "zarf-git-read-user"},
+	}
+
+	expected := map[string]string{
+		"###ZARF_GIT_AUTH_PULL###": "**sanitized**",
+		"###ZARF_GIT_AUTH_PUSH###": "**sanitized**",
+		"###ZARF_GIT_PULL###": "zarf-git-read-user",
+		"###ZARF_GIT_PUSH###": "zarf-git-user",
+		"###ZARF_REGISTRY###": "127.0.0.1:31999",
+	}
+
+	output := getSanitizedTemplateMap(input)
+	require.Equal(t, expected, output)
```

Comment:
There's clean data here but I'd recommend using a test table for this approach and running checks for the simplest test cases as well as the complex one. Empty values are great to test for as well. What if we pass in a nil map, empty map, or a map with a real key nil entry `variables.TextTemplate`?

```suggestion
tt := struct{
	name string
	in map[string]*variables.TextTemplate
	expect map[string]string
}{
	{
		// TODO cases: can nils input, empty in
	},
	{
		// TODO case: can handle map with non-sensitive value
	},
	{
		name: "can redact a sensitive value"
		in: map[string]*variables.TextTemplate{
			"###ZARF_GIT_AUTH_PULL###": {Sensitive: true, Value: "secret1"},
		},
		expect: map[string]string{
			"###ZARF_GIT_AUTH_PULL###": "**sanitized**",
		},
	},
	{
		name: "can handle both sensitive and non-sensitive values at once"
		in: map[string]*variables.TextTemplate{
			"###ZARF_GIT_AUTH_PULL###": {Sensitive: true, Value: "secret1"},
			"###ZARF_GIT_AUTH_PUSH###": {Sensitive: true, Value: "secret2"},
			"###ZARF_REGISTRY###":      {Sensitive: false, Value: "127.0.0.1:31999"},
			"###ZARF_GIT_PUSH###":      {Sensitive: false, Value: "zarf-git-user"},
			"###ZARF_GIT_PULL###":      {Sensitive: false, Value: "zarf-git-read-user"},
		},
		expect: map[string]string{
			"###ZARF_GIT_AUTH_PULL###": "**sanitized**",
			"###ZARF_GIT_AUTH_PUSH###": "**sanitized**",
			"###ZARF_GIT_PULL###": "zarf-git-read-user",
			"###ZARF_GIT_PUSH###": "zarf-git-user",
			"###ZARF_REGISTRY###": "127.0.0.1:31999",
		},
	}, 
}

for _, tc := range tt {
    t.Run(tc.name, func(t *testing.T) {
        // TODO Requires here 
    })
}
```
(psuedocode)

Reply:
Good call! It would panic before if variables.TextTemplate was nil, fixed that and added test cases.